sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -313,6 +313,14 @@ enum class graph_support_level {
   emulated
 };
 
+enum class node_type {
+			empty = 0,
+			kernel,
+			memcpy,
+			memset,
+			memfill,
+};
```

Comment:
> Is this list mean to be exhaustive? Other node types we could have are ext_oneapi_barrier, prefetch, memadvise.
It should be yes, I've filled out the enum now.

> You're point about subgraphs is also interesting, there's also the kernel fusion optimization.
> I think we need to decide if this exposes what nodes to user submitted or how we're actually representing them in the runtime.

I think that generally I favour that we should represent the graph as the user submitted it, unless we are explicitly doing optimizations (which we don't yet) which would affect our internal structure of the graph. If the optimizations don't actually require changing our internal representation though then that shouldn't be an issue, since these would likely occur at finalize time anyway.

For subgraphs I think we could either make the current implementation approach explicit in the specification so that the user knows what to expect, or change the wording to reflect that a `node_type::subgraph` node is added which represents the subgraph as part of the parent graph. I don't particularly have a preference either way, I'm not sure what would make most sense to users.